plugins/modules/aws_glue_job.py

```diff
@@ -303,11 +345,25 @@ def create_or_update_glue_job(connection, module, glue_job):
         except (BotoCoreError, ClientError) as e:
             module.fail_json_aws(e)
 
+    # handle resource tags
+    resource_arn = _get_glue_job_arn(module, params['Name'])
+
+    new_tags = module.params.get('tags')
+    existing_tags = connection.get_tags(ResourceArn=resource_arn).get('Tags')
+    tags_to_add, tags_to_remove = compare_aws_tags(existing_tags, new_tags if new_tags else {}, module.params.get('purge_tags'))
+
+    if tags_to_remove:
+        connection.untag_resource(ResourceArn=resource_arn, TagsToRemove=tags_to_remove)
+        changed = True
+    if tags_to_add:
+        connection.tag_resource(ResourceArn=resource_arn, TagsToAdd=tags_to_add)
+        changed = True
```

Comment:
```suggestion
    if new_tags is not None:
        existing_tags = connection.get_tags(ResourceArn=resource_arn).get('Tags')
        tags_to_add, tags_to_remove = compare_aws_tags(existing_tags, new_tags, module.params.get('purge_tags'))

        if tags_to_remove:
            connection.untag_resource(ResourceArn=resource_arn, TagsToRemove=tags_to_remove)
            changed = True

        if tags_to_add:
            connection.tag_resource(ResourceArn=resource_arn, TagsToAdd=tags_to_add)
            changed = True
```

The trouble is that `purge_tags` **defaults** to true, this means that if you pass no tags, by default you're going to remove the tags, even if neither tags nor purge_tags is explicitly passed.  You're not catching this in the tests right now because you set `purge_tags: false` in the module_defaults.

The minimal tweak here is to wrap the new code in `if new_tags is not None:`

However, I find it easier to follow if you split the changing of each optional parameter off into its own function, then you can do:

```
changed |= update_some_attribute(connection, module)
```

```
def update_some_attribute(connection, module):
    if module.params.get('some_attribute') is None:
        return False
    if <expected_state == current_state>:
        return False
    if module.check_mode:
        return True
    <Make the change>
    return True
```

This makes each function less of a rats nest and the code can start documenting itself (you also avoid triggering pylint's `too-many-nested-blocks` warning.